sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/CompactBlockProcessor.kt

```diff
@@ -559,8 +559,12 @@ class CompactBlockProcessor internal constructor(
     ): Int = withContext(IO) {
         var skipped = 0
         val aboveHeight = startHeight
-        twig("Clearing utxos above height $aboveHeight", -1)
-        rustBackend.clearUtxos(tAddress, aboveHeight)
+        // TODO(str4d): We no longer clear UTXOs here, as rustBackend.putUtxo uses an upsert. However, because the data
+        //  we habe here is just the list of _current_ UTXOs, this means that now-spent notes are not marked as such.
+        //  This will be almost fully addressed by https://github.com/zcash/librustzcash/pull/677 which marks them as
```

Comment:
Before we deploy an SDK to users, do we need to do another pull request that includes this librustzcash update?

Reply:
Yes, otherwise shielding won't work correctly. But I opened this PR to unblock you in the meantime.